File: readnwritecapitalize.py
import os
import re

# Open the input file for reading
with open("haha.txt", "r") as input_file:
    content = input_file.readlines()

# Convert all the text to lowercase
content = [line.lower() for line in content]

# Capitalize one word (case-insensitive) (for example, the word "doit")
word_to_capitalize = "doit"
capitalized_word = word_to_capitalize.capitalize()
content = [re.sub(word_to_capitalize, capitalized_word, line, flags=re.IGNORECASE) for line in content]

# Create the output file for writing
with open("output.txt", "w") as output_file:
    output_file.writelines(content)

File: readnwritecapitalize.js
const fs = require('fs');
const util = require('util');

// Promisify the readFile and writeFile functions
const readFile = util.promisify(fs.readFile);
const writeFile = util.promisify(fs.writeFile);

async function processFile() {
  try {
    // Read the entire content of the input file
    const content = await readFile('haha.txt', 'utf8'); // Read the file as a string

    // Split the content into lines
    const lines = content.split('\n');

    // Convert all the text to lowercase
    const lowercaseContent = lines.map(line => line.toLowerCase());

    // Capitalize the word 'doit' (case-insensitive)
    const wordToCapitalize = 'doit';
    const modifiedContent = lowercaseContent.map(line =>
      line.replace(new RegExp(wordToCapitalize, 'gi'), wordToCapitalize.charAt(0).toUpperCase() + wordToCapitalize.slice(1))
    );

    // Write the modified content to the output file
    await writeFile('output.txt', modifiedContent.join('\n'));
    console.log('File processing completed successfully.');
  } catch (err) {
    console.error('Error processing file:', err);
  }
}

// Call the processFile function
processFile();
